src/components/profiles-modals/ImportProfileModal.vue

```diff
@@ -0,0 +1,378 @@
+    downloadProgressCallback(progress: number, modName: string, status: number, err: R2Error | null) {
+        if (status == StatusEnum.FAILURE) {
+            this.isProfileBeingImported = false;
+            if (err instanceof R2Error) {
+                this.$store.commit('error/handleError', err);
+            }
+        } else if (status == StatusEnum.PENDING) {
+            this.percentageImported = Math.floor(progress);
+        }
+    }
+
+    // Called from the downloadImportedProfileMods function
+    async downloadCompletedCallback(comboList: ThunderstoreCombo[], modList: ExportMod[], callback?: () => void) {
+        let keepIterating = true;
+        for (const comboMod of comboList) {
+            if (!keepIterating) {
+                return;
+            }
+            const installResult: R2Error | ManifestV2 = await this.installModAfterDownload(comboMod.getMod(), comboMod.getVersion());
+            if (installResult instanceof R2Error) {
+                this.$store.commit('error/handleError', installResult);
+                keepIterating = false;
+                this.isProfileBeingImported = false;
+                return;
+            }
+            for (const imported of modList) {
+                if (imported.getName() == comboMod.getMod().getFullName() && !imported.isEnabled()) {
+                    await ProfileModList.updateMod(installResult, this.activeProfile, async (modToDisable: ManifestV2) => {
+                        // Need to enable temporarily so the manager doesn't think it's re-disabling a disabled mod.
+                        modToDisable.enable();
+                        await ProfileInstallerProvider.instance.disableMod(modToDisable, this.activeProfile);
+                        modToDisable.disable();
+                    });
+                }
+            }
+        }
+        if (callback !== undefined) {
+            callback();
+        }
+        this.isProfileBeingImported = false;
+    }
+
+    // Called when the name for the imported profile is given and confirmed by the user.
+    async createProfile() {
+        const safeName = this.makeProfileNameSafe(this.newProfileName);
+        if (safeName === '') {
+            return;
+        }
+        try {
+            await this.$store.dispatch('profiles/addProfile', safeName);
+            document.dispatchEvent(new CustomEvent("created-profile", {detail: safeName}));
+        } catch (e) {
+            const err = R2Error.fromThrownValue(e, 'Error importing a profile');
+            this.$store.commit('error/handleError', err);
+        }
+    }
+
+    // Called when the profile to update is selected and confirmed by the user.
+    updateProfile() {
+        document.dispatchEvent(new CustomEvent("created-profile", {detail: this.activeProfileName}));
+    }
+}
+
+</script>
+<template>
+    <ModalCard :is-active="isOpen" @close-modal="closeModal">
+
+        <template v-slot:header v-if="activeStep === 'IMPORT_UPDATE_SELECTION'">
```

Comment:
This wasn't quite what I meant with my earlier comment. This if-else-else makes it difficult to find out what the actual rendered output for each step is. I was thinking more in line having something like:

```vue
<ModalCard :is-active="isOpen" @close-modal="closeModal">
  <template v-slot:header v-if="activeStep === 'IMPORT_UPDATE_SELECTION'">
    <!-- all IMPORT_UPDATE_SELECTION markup here -->
  </template>
  
  <template v-slot:header v-if="activeStep === 'FILE_CODE_SELECTION'">
    <!-- all FILE_CODE_SELECTIONmarkup here -->
  </template>
  
  <!-- And so on -->
</ModalCard>
```